src/vm/memory/memory.zig

```diff
@@ -476,12 +476,28 @@ pub const Memory = struct {
         try self.validation_rules.put(@intCast(segment_index), rule);
     }
 
-    /// Marks a `MemoryCell` as accessed at the specified relocatable address.
+    /// Marks a `MemoryCell` as accessed at the specified relocatable address within the memory segment.
+    ///
+    /// # Description
+    /// This function marks a memory cell as accessed at the provided relocatable address within the memory segment.
+    /// It enables tracking and management of memory access for effective memory handling within the Cairo VM.
+    ///
     /// # Arguments
-    /// - `address` - The relocatable address to mark.
+    /// - `address` - The relocatable address to mark as accessed within the memory segment.
+    ///
+    /// # Safety
+    /// This function assumes correct usage and does not perform bounds checking. It's the responsibility of the caller
+    /// to ensure that the provided `address` is within the valid bounds of the memory segment.
     pub fn markAsAccessed(self: *Self, address: Relocatable) void {
         const segment_index: usize = @intCast(if (address.segment_index < 0) -(address.segment_index + 1) else address.segment_index);
-        (if (address.segment_index < 0) &self.temp_data else &self.data).items[segment_index].items[address.offset].?.markAccessed();
+        var data = if (address.segment_index < 0) &self.temp_data else &self.data;
+
+        if (segment_index < data.items.len) {
+            if (address.offset < data.items[segment_index].items.len) {
+                if (data.items[segment_index].items[address.offset] != null)
+                    data.items[segment_index].items[address.offset].?.is_accessed = true;
```

Comment:
hey I think we need to remove the function `markAccessed` completely since now we don't use it in the one place we used it